app/actions/client.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import httpx
+import pydantic
+import stamina
+
+from typing import Optional
+from app.services.state import IntegrationStateManager
+
+
+logger = logging.getLogger(__name__)
+state_manager = IntegrationStateManager()
+
+
+class Farm(pydantic.BaseModel):
+    id: str = pydantic.Field(alias='_id')
+    name: str
+    nif: Optional[str]
+    rega: Optional[str]
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class FarmLocation(pydantic.BaseModel):
+    location: tuple[float, float] = pydantic.Field(alias='_location')
+    time: str = pydantic.Field(alias='_time')
```

Comment:
I'm weary of keeping track of timestamps as strings. You might consider taking advantage of pydantic's parsing and validation for this property and some others (ex. `start` and `stop` in the PullFarmObservationsConfig model).
Keeping the parsing and validation in one place, and doing it consistently will save us from mysterious errors int he future.